Modules/Quotas/src/Quotas.Application/Identities/Queries/ListQuotasForIdentity/Handler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Quotas.Application.DTOs;
+using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
+using Backbone.Modules.Quotas.Domain.Metrics;
+using MediatR;
+
+namespace Backbone.Modules.Quotas.Application.Identities.Queries.ListQuotasForIdentity;
+
+public class Handler : IRequestHandler<ListQuotasForIdentityQuery, ListQuotasForIdentityResponse>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly MetricCalculatorFactory _metricCalculatorFactory;
+    private readonly IMetricsRepository _metricsRepository;
+    private readonly IdentityAddress _identityAddress;
+
+    public Handler(IUserContext userContext, IIdentitiesRepository identitiesRepository, IMetricsRepository metricsRepository, MetricCalculatorFactory metricCalculatorFactory)
+    {
+        _identityAddress = userContext.GetAddress();
+        _identitiesRepository = identitiesRepository;
+        _metricsRepository = metricsRepository;
+        _metricCalculatorFactory = metricCalculatorFactory;
+    }
+
+    public async Task<ListQuotasForIdentityResponse> Handle(ListQuotasForIdentityQuery request, CancellationToken cancellationToken)
+    {
+        var identity = await _identitiesRepository.Find(_identityAddress, cancellationToken) ?? throw new NotFoundException(nameof(Identity));
```

Comment:
I don't think we should throw a NotFoundException here, as this will result in a 404, meaning that it's the user's fault. But actually, since we don't make use of any user input except for the token (which is generated by us), the user can't do anything about it.
So you can just throw an Exception with a descriptive error message. This will result in a 500, which is much more fitting.

Reply:
Why did you resolve this comment? You didn't change anything.

Reply:
No idea to be honest :( It's now taken care of, hopefully message is of good quality as well.